engine/MapFunctions.ts

```diff
@@ -1,37 +1,42 @@
+import { ComponentJson } from '@etherealengine/common/src/interfaces/SceneInterface'
+import { LoadGLTF } from '@etherealengine/engine/src/assets/functions/LoadGLTF'
+import { defaultAvatarHalfHeight } from '@etherealengine/engine/src/avatar/functions/spawnAvatarReceptor'
 import { DebugNavMeshComponent } from '@etherealengine/engine/src/debug/DebugNavMeshComponent'
-import { Entity } from '@etherealengine/engine/src/ecs/classes/Entity'
-import { Object3D, Group, Mesh } from 'three'
 import { Engine } from '@etherealengine/engine/src/ecs/classes/Engine'
+import { EngineState } from '@etherealengine/engine/src/ecs/classes/EngineState'
+import { Entity } from '@etherealengine/engine/src/ecs/classes/Entity'
+import { addComponent, getAllComponents, getComponent, hasComponent } from '@etherealengine/engine/src/ecs/functions/ComponentFunctions'
 import { NavMeshComponent } from '@etherealengine/engine/src/navigation/component/NavMeshComponent'
-import { MapAction, mapReducer } from './MapReceptor'
-import { MapComponent } from './MapComponent'
-import { getPhases, startPhases } from './functions/PhaseFunctions'
-import { LoadGLTF } from '@etherealengine/engine/src/assets/functions/LoadGLTF'
-import { avatarHalfHeight } from '@etherealengine/engine/src/avatar/functions/createAvatar'
-import { Text } from 'troika-three-text'
 import { Object3DComponent } from '@etherealengine/engine/src/scene/components/Object3DComponent'
-import { ComponentJson } from '@etherealengine/common/src/interfaces/SceneInterface'
-import { isClient } from '@etherealengine/engine/src/common/functions/isClient'
 import { registerSceneLoadPromise } from '@etherealengine/engine/src/scene/functions/SceneLoading'
-import { EntityNodeComponent } from '@etherealengine/engine/src/scene/components/EntityNodeComponent'
-import { addChildFast, setPosition } from './util'
+import { getState } from '@etherealengine/hyperflux'
+
 import { debounce } from 'lodash'
+import { Group, Mesh, Object3D } from 'three'
+import { Text } from 'troika-three-text'
+
+import { isClient } from '@etherealengine/engine/src/common/functions/getEnvironment'
+import { MapComponent, MapComponentType } from './MapComponent'
+import { MapAction, mapReducer } from './MapReceptor'
+import { getPhases, startPhases } from './functions/PhaseFunctions'
+import { getStartCoords } from './getStartCoords'
+import { addChildFast, setPosition } from './util'
 
 export const SCENE_COMPONENT_MAP = 'map'
 export const SCENE_COMPONENT_MAP_DEFAULT_VALUES = {}
 
 export const deserializeMap = (entity: Entity, json: ComponentJson<MapComponentType>) => {
   if (isClient) {
-    registerSceneLoadPromise(createMap(entity, json.props))
-    if (Engine.isEditor) getComponent(entity, EntityNodeComponent)?.components.push(SCENE_COMPONENT_MAP)
+    registerSceneLoadPromise(createMap(entity, json.props as MapComponentType))
```

Comment:
registerSceneLoadPromise no longer exists, we simply need to add SceneAssetPendingTagComponent